src/components/StudyPane/InfoPane/Motif/RootBlock.tsx

```diff
@@ -66,36 +53,59 @@ export const RootBlock = ({
     setSelected(hasChildren);
   }, [ctxSelectedHebWords, descendants]);
 
-  const handleClick = (e: React.MouseEvent) => {
-    setSelected(prevState => !prevState);
-
-    let updatedSelectedHebWords = [...ctxSelectedHebWords];
-    if (!selected) {
-      updatedSelectedHebWords = ctxSelectedHebWords.concat(descendants);
-    } else {
-      descendants.forEach((dsd) => {
-        updatedSelectedHebWords.splice(updatedSelectedHebWords.indexOf(dsd), 1)
-      })
-    }
-    ctxSetSelectedHebWords(updatedSelectedHebWords);
-    ctxSetNumSelectedWords(updatedSelectedHebWords.length);
-  };
-
   useEffect(() => {
-    if (ctxColorAction !== ColorActionType.none && selected) {
-      if (ctxColorAction === ColorActionType.colorFill && ctxSelectedColor) {
-        setColorFillLocal(ctxSelectedColor);
-      } else if (ctxColorAction === ColorActionType.borderColor && ctxSelectedColor) {
-        setBorderColorLocal(ctxSelectedColor);
-      } else if (ctxColorAction === ColorActionType.textColor && ctxSelectedColor) {
-        setTextColorLocal(ctxSelectedColor);
-      } else if (ctxColorAction === ColorActionType.resetColor) {
-        setColorFillLocal(DEFAULT_COLOR_FILL);
-        setBorderColorLocal(DEFAULT_BORDER_COLOR);
-        setTextColorLocal(DEFAULT_TEXT_COLOR);
+    if (selected){
+      const matchedColorScheme = matchExpandedWordsColorScheme()
```

Comment:
What is this change for?

Reply:
The latest version looks like 
`if (selected){
      const matchedFillColor = matchFillColor();
      const matchedTextColor = matchTextColor();
      const matchedBorderColor = matchBorderColor();`

it aids in being able to retain the custom colours of each wordBlock further down in the useEffect function. If there is a match on any categories, they retain it, if not each wordBlock retain their own color scheme except for the one that is being changed: 
` if (ctxColorAction !== ColorActionType.none && selected) {
        if (ctxColorAction === ColorActionType.colorFill && ctxSelectedColor) {
          
          setColorFillLocal(ctxSelectedColor);
          colorObject.colorFill = ctxSelectedColor;
          colorObject.textColor = matchedTextColor && rootBlockColor?rootBlockColor.textColor!==''?rootBlockColor.textColor:'':''
          colorObject.borderColor = matchedBorderColor && rootBlockColor?rootBlockColor.borderColor!==''?rootBlockColor.borderColor:'':'';
          let descendantWordIds: number[] = [];
          descendants.forEach((word) => {
            word.colorFill = colorObject.colorFill;
            word.textColor = word.textColor
            word.borderColor = word.borderColor;
            descendantWordIds.push(word.id)
          });`
this just shows the colour fill being adjusted, and further down in the useEffect, similar calls are being made to the other colorActions